src/grammar/sequence.rs

```diff
@@ -0,0 +1,137 @@
+use crate::{safe_loop, Parser};
+use source_gen::{lexer::TokenKind, syntax::SyntaxKind, T};
+
+use super::parse_ident;
+
+#[allow(unused)]
+pub(crate) fn parse_sequence(p: &mut Parser) {
+    p.start(SyntaxKind::SequenceStmt);
+    p.expect(T![create]);
+    p.expect(T![sequence]);
+    if p.eat(T![if]) {
+        p.expect(T![not]);
+        p.expect(T![exists]);
+    }
+    parse_ident(p, 1..2);
+    if p.eat(T![sharing]) {
+        p.expect(T![=]);
+        p.expect_one_of(&[T![metadata], T![data], T![none]]);
+    }
+    parse_sequence_params(p);
+    p.eat(T![;]);
+    p.finish();
+}
+
+pub(crate) fn parse_sequence_params(p: &mut Parser) {
+    p.start(SyntaxKind::SequenceParameters);
+    // parse until no longer can lol
+    safe_loop!(p, {
+        match p.current() {
+            T![increment] => {
+                p.expect(T![increment]);
+                p.expect(T![by]);
+                p.expect(T![int_literal]);
+            }
+            T![start] => {
+                p.expect(T![start]);
+                p.expect(T![with]);
+                p.expect(T![int_literal]);
+            }
+            T![maxvalue] | T![minvalue] | T![cache] => {
+                p.expect_one_of(&[T![maxvalue], T![minvalue], T![cache]]);
```

Comment:
Since we know that we are at one of these tokens, we can just bump.
```suggestion
                p.bump_any();
```